<?xml version="1.0"?>
<launch>
  <!-- Camera driver setting -->
  <!-- <include file="$(find usb_cam)/"/> -->
  
  <!-- lidar driver setting -->
  <!-- <include file="$(find lidar name)/"/> -->
    <include file="$(find robot_bringup)/launch/robot_bringup.launch"/>

  <node name="Deepsort" pkg="smart_pkg" type="ds_node.py"  output="screen">
    <!-- PID controller setting -->

    <!-- if angref == True -> x: linear y: angular -->
    <param name="angref" value="True"/>

    <param name="kp_x" value="1.0"/>
    <param name="ki_x" value="0.0"/>
    <param name="kd_x" value="0.0"/>

    <param name="kp_y" value="1.0"/>
    <param name="ki_y" value="0.0"/>
    <param name="kd_y" value="0.0"/>

    <!-- control_period setting -->
    <param name="control_period" value="2.0"/>

    <!-- deepsort params -->
    <param name="track_model_path" value="/home/nvidia/catkin_ws/src/smart_pkg/include/deep_sort/deep/checkpoint/ckpt.t7"/>
    <param name="detect_age" value="3"/>
    
    <!-- put YOLO model path -->
    <param name="detect_model_path" value="yolov8n.pt"/>
  </node>

</launch>